packages/components/src/mobile/audio-player/index.native.js

```diff
@@ -119,7 +118,9 @@ function Player( {
 	let title = '';
 	let extension = '';
 
-	if ( fileName ) {
+	if ( src ) {
+		const decodedURI = decodeURI( src );
```

Comment:
@ceyhun where does `decodeURI ` come from exactly?

Reply:
I also just saw a safe version of decodeURI https://github.com/WordPress/gutenberg/blob/004a32ec09d5b789be694d85725fe49cca385f08/packages/url/src/safe-decode-uri.js can that be helpful here in anyway?

Reply:
Oh @ceyhun `decodeURI` is a built-in Javascript function :) I just realized that.

Reply:
It's built-in in JS, you can check the reference [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects#function_properties) for some others. BTW I'm not really happy parsing the file name and extension from the media src here, I would appreciated any other ideas or suggestions 🙏

Reply:
>I also just saw a safe version of decodeURI 004a32e/packages/url/src/safe-decode-uri.js can that be helpful here in anyway?

Good catch! Should better use this one 👍

Reply:
> It's built-in in JS, you can check the reference here for some others.

okay understood. Thanks for the reference.

> . BTW I'm not really happy parsing the file name and extension from the media src here, I would appreciated any other ideas or suggestions 🙏

Could you expound on this a bit? Do you mean that you aren't happy with the location it is in? Or you are not sure if the output/result of this parsing will be reliable and user friendly? 

Another question I wanted to ask is could we check to see if the media title is present and we use that as the title and if it's not present we use the filename that's part of the URI. Let me know if this helps in anyway. I am looking at the media object and the overall logic to see if I can find any more clues to assist.

Reply:
I am probably misinterpreting this @ceyhun is it that when the file name and extension are being parsed sometimes it's missing which is leading to the title of the audio file being used as the extension?

Reply:
> Or you are not sure if the output/result of this parsing will be reliable and user friendly?

Closer to this ☝️ 

> Another question I wanted to ask is could we check to see if the media title is present and we use that as the title and if it's not present we use the filename that's part of the URI. Let me know if this helps in anyway. I am looking at the media object and the overall logic to see if I can find any more clues to assist.
 
Was using the media title before, but it was not there when selecting from media library.

> I am probably misinterpreting this @ceyhun is it that when the file name and extension are being parsed sometimes it's missing the extension which is leading to the title of the audio file being used as the extension?

If we use `src` which is the URI of the file when picking from the phone's local storage and URL of the file picking from media library. In both cases for iOS and Android `src` had a filename and extension in the URI+URL. But the filename isn't always the same as the file title assigned to the file when it's uploaded to the remote from local storage.